fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java

```diff
@@ -302,17 +303,53 @@ public Cost visitPhysicalHashJoin(
             }
             // TODO: since the outputs rows may expand a lot, penalty on it will cause bc never be chosen.
             // will refine this in next generation cost model.
+            if (isStatsUnknown(physicalHashJoin, buildStats, probeStats)) {
+                // forbid broadcast join when stats is unknown
+                return CostV1.of(rightRowCount * buildSideFactor + 1 / leftRowCount,
+                        rightRowCount,
+                        0
+                );
+            }
             return CostV1.of(leftRowCount + rightRowCount * buildSideFactor + outputRowCount * probeSideFactor,
                     rightRowCount,
                     0
             );
         }
+        if (isStatsUnknown(physicalHashJoin, buildStats, probeStats)) {
+            return CostV1.of(rightRowCount + 1 / leftRowCount,
+                    rightRowCount,
+                    0);
+        }
         return CostV1.of(leftRowCount + rightRowCount + outputRowCount,
                 rightRowCount,
                 0
         );
     }
 
+    private boolean isStatsUnknown(PhysicalHashJoin<? extends Plan, ? extends Plan> join,
```

Comment:
PhysicalHashJoin -> AbstractPhysicalJoin
use AbstractPhysicalJoin, and this function can be used both for hash join and nested loop join